upload/library/SV/WarningImprovements/XenForo/Model/Warning.php

```diff
@@ -928,13 +1035,31 @@ protected function _userWarningPointsIncreased(
 
     public function triggerWarningAction($userId, array $action)
     {
+        if ($action['sv_warning_category_id'] !== 0)
```

Comment:
I'm not sure triggerWarningAction will be consistently called from _userWarningPointsIncreased for a warning category.

In _userWarningPointsIncreased, ```$newPoints``` and ```$oldPoints``` are terms of global warning points, which makes the loop inaccurate for triggering the per-category warnings:
```
		foreach ($actions AS $action)
		{
			if ($action['points'] <= $oldPoints)
			{
				continue;
			}
			else if ($action['points'] > $newPoints)
			{
				continue; // no trigger yet
			}
			....
```